fixes 1611: adds page for boilerplate config  and clarifies profiles (#1644)
hyperledger/besu-docs: docs/public-networks/how-to/configure-besu/profile.md

---
title: Use a profile 
sidebar_position: 1
tags:
  - public networks
  - private networks
---

# Use a profile 

> Load a profile using the [`--profile` CLI option](../../reference/cli/options.md#profile).
> 
> :::note
> Run `./besu --help` to view all available profiles.
> :::
> 
> You can optionally use profiles to extend Besu's configuration. Combined with the 
> [boilerplate configuration](index.md#besu-defaults), profiles simplify the process of applying 
> viable defaults. Besu's pre-configured profiles optimize for the following use cases:

Reviewer: ```suggestion
You can use profiles to extend Besu's [default configuration](index.md#default-configuration), using the [`--profile`](../../reference/cli/options.md#profile) option.

Profiles simplify the process of configuring Besu for specific use cases. Besu provides the following pre-configured profiles:
```

Suggested re-wording for simplicity.

- [Minimalist staker profile](#minimalist-staker-profile)
- [Staker profile](#staker-profile)
- [Enterprise/Private profile](#enterpriseprivate-profile)

Alternatively, you can customize and [load external profiles](#load-external-profiles).

> 

Reviewer: ```suggestion

:::note
Run `./besu --help` to view all available profiles.
:::

```

This note is not as important, can be moved to end of intro.

:::note

> A configuration explicitly set in the configuration file or command line will 
> [override the same option/s](index.md#configuration-order-of-precedence) set in the profile.

Reviewer: ```suggestion
A configuration option specified in the configuration file or on the command line will 
[override the same option](index.md#configuration-order-of-precedence) set in the profile.
```

:::

## Minimalist staker profile

[`--profile=MINIMALIST_STAKER`](../../reference/cli/options.md#profile) is optimized for stakers who 
want to maximize their hardware value but don't want to serve full sets of data to their peers, See the
[minimalist staker profile on GitHub](https://github.com/hyperledger/besu/blob/main/config/src/main/resources/profiles/minimalist-staker.toml)
for the custom settings.

## Staker profile

[`--profile=STAKER`](../../reference/cli/options.md#profile) is optimized for stakers who want to 
maximize their hardware value while also serving full sets of data to their peers. See the
[staker profile on GitHub](https://github.com/hyperledger/besu/blob/main/config/src/main/resources/profiles/staker.toml)
for the custom settings.

## Enterprise/Private profile

> `enterprise` and `private` are aliases for the same profile. [`--profile=PRIVATE` / `--profile=ENTERPRISE`](../../reference/cli/options.md#profile) 

Reviewer: ```suggestion
`ENTERPRISE` and `PRIVATE` are aliases for the same profile. [`--profile=PRIVATE` / `--profile=ENTERPRISE`](../../reference/cli/options.md#profile) 
```

supports private network operators and enterprises by handling specific use cases that apply to 
private network operators. See the [enterprise/private profile on 
GitHub](https://github.com/hyperledger/besu/blob/main/config/src/main/resources/profiles/enterprise-private.toml)
for the custom settings.

> 
> To use the enterprise/private profile, run Besu with
> [`--profile`](../../reference/cli/options.md#profile) set to `--profile=PRIVATE` or `--profile=ENTERPRISE`, 
> and use [`sync-mode=FULL`](../../reference/cli/options.md#sync-mode) 
> and [`data-storage-format=FOREST`](../../reference/cli/options.md#data-storage-format).

Reviewer: ```suggestion
When using this profile, set [`--sync-mode=FULL`](../../reference/cli/options.md#sync-mode) 
and [`--data-storage-format=FOREST`](../../reference/cli/options.md#data-storage-format).
```

> ## Load external profiles

Author: @joaniefromtheblock does adding this functionality require update to "what's new"?

Reviewer 2: I don't think we add a "what's new" for Besu. I might be wrong about that, but I think new additions are just added to the changeling by Eng

Reviewer 3: correct, we shouldn't add whats new to the docs as it will be a headache to maintain.
